extensions/scarb-cairo-execute/src/main.rs

```diff
@@ -0,0 +1,338 @@
+        Program::new(
+            entrypoint.builtins.clone(),
+            data,
+            Some(entrypoint.offset),
+            hints,
+            Default::default(),
+            Default::default(),
+            vec![],
+            None,
+        )
+    }
+    .with_context(|| "Failed setting up program.")?;
+
+    let mut hint_processor = CairoHintProcessor {
+        runner: None,
+        user_args: vec![vec![Arg::Array(
+            args.run
+                .arguments
+                .iter()
+                .map(|v| Arg::Value(v.into()))
+                .collect(),
+        )]],
+        string_to_hint,
+        starknet_state: Default::default(),
+        run_resources: Default::default(),
+        syscalls_used_resources: Default::default(),
+        no_temporary_segments: false,
+    };
+
+    let cairo_run_config = CairoRunConfig {
+        trace_enabled: true,
+        relocate_mem: false,
+        layout: LayoutName::all_cairo,
+        proof_mode: args.run.proof_mode,
+        secure_run: None,
+        allow_missing_builtins: Some(true),
+        ..Default::default()
+    };
+
+    let mut runner = cairo_run_program(&program, &cairo_run_config, &mut hint_processor)
+        .map_err(|e| anyhow::anyhow!("Cairo program run failed: {e:?}"))?;
+
+    if args.run.print_outputs {
+        let mut output_buffer = "Program Output:\n".to_string();
+        runner.vm.write_output(&mut output_buffer)?;
+        print!("{output_buffer}");
+    }
+
+    let directory_path = Path::new(BASE_OUTPUT_DIRECTORY);
+    create_output_dir(directory_path)?;
+    let base_path = create_execution_directory(directory_path)?;
+
+    if !args
+        .run
+        .output
+        .unwrap_or(OutputFormat::Standard)
+        .is_cairo_pie()
+    {
+        let trace_file_path = base_path.join(TRACE_FILENAME);
+
+        let relocated_trace = runner
+            .relocated_trace
+            .as_ref()
+            .with_context(|| "Trace not relocated.")?;
+        let mut writer = FileWriter::new(3 * 1024 * 1024, &trace_file_path)?;
+        cairo_run::write_encoded_trace(relocated_trace, &mut writer)?;
+        writer.flush()?;
+
+        let memory_file_path = base_path.join(MEMORY_FILENAME);
+        let mut writer = FileWriter::new(5 * 1024 * 1024, &memory_file_path)?;
+        cairo_run::write_encoded_memory(&runner.relocated_memory, &mut writer)?;
+        writer.flush()?;
+
+        let air_public_file_path =
+            create_incremental_file_in_dir(&base_path, Some(AIR_PUBLIC_FILE.1), AIR_PUBLIC_FILE.0)?;
+        let json = runner.get_air_public_input()?.serialize_json()?;
+        fs::write(&air_public_file_path, json)?;
+
+        let air_private_file_path = create_incremental_file_in_dir(
+            &base_path,
+            Some(AIR_PRIVATE_FILE.1),
+            AIR_PRIVATE_FILE.0,
+        )?;
+        let absolute = |path_buf: PathBuf| {
+            path_buf
+                .as_path()
+                .canonicalize()
```

Comment:
`cargo clippy` warns agains using `canonicalize()`, should i use `fsx::canonicalize_utf8` does it apply in this case? please advise

Reply:
Yes, you should. `fsx` is basically our wrapper over path manipulation functions that handles some edge-cases that we have run into in the past. The clippy lint you see has been added by us here https://github.com/software-mansion/scarb/blob/f1e8abb65ee21efb2c813ff95b8d2ecd440d940c/clippy.toml#L1